modules/core/04-channel/keeper/upgrade.go

```diff
@@ -95,8 +93,18 @@ func (k Keeper) WriteUpgradeInitChannel(
 	emitChannelUpgradeInitEvent(ctx, portID, channelID, upgradeSequence, channelUpgrade)
 }
 
+// Restorable defines an interface for an ibc module which can be restored.
+// All upgradable modules should implement this interface.
+type Restorable interface {
+	OnChanUpgradeRestore(
+		ctx sdk.Context,
+		portID,
+		channelID string,
+	) error
```

Comment:
I thought we agreed not to return an error

Reply:
yes! This PR was opened before that change was made.